File: frontend/apps/ios/Pantopus/Features/Mailbox/MailDetail/Variants/Components/BidCard.swift
//
//  BidCard.swift
//  Pantopus
//
//  The bid headline card for the A17.6 Gig mail body — large amount on
//  the left, unit + ETA, an expiry / "Locked in" pill on the right, and
//  the bidder's multi-line message below. Tinted with the gig accent
//  (orange) so it reads as the focal surface. Mirrors the Android
//  `BidCard.kt` shape so the two platforms render the same focal card.
//

import SwiftUI

@MainActor
public struct BidCard: View {
    private let bid: GigDetailDTO.Bid
    private let isAccepted: Bool

    public init(bid: GigDetailDTO.Bid, isAccepted: Bool) {
        self.bid = bid
        self.isAccepted = isAccepted
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: Spacing.s0) {
            amountRow
                .padding(Spacing.s3)
            Rectangle()
                .fill(Theme.Color.handyman.opacity(0.3))
                .frame(height: 1)
                .accessibilityHidden(true)
            messageBlock
                .padding(Spacing.s3)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Theme.Color.handyman.opacity(0.06))
        .clipShape(RoundedRectangle(cornerRadius: Radii.xl))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.xl)
                .stroke(Theme.Color.handyman.opacity(0.4), lineWidth: 1.5)
        )
        .accessibilityElement(children: .contain)
        .accessibilityIdentifier("gigBidCard")
    }

    private var amountRow: some View {
        HStack(alignment: .top, spacing: Spacing.s3) {
            VStack(alignment: .leading, spacing: Spacing.s1) {
                Text("BID AMOUNT")
                    .pantopusTextStyle(.overline)
                    .foregroundStyle(Theme.Color.handyman)
                HStack(alignment: .firstTextBaseline, spacing: Spacing.s1) {
                    Text("$\(bid.amount)")
                        .font(.system(size: 34, weight: .heavy))
                        .foregroundStyle(Theme.Color.appTextStrong)
                    Text("· \(bid.unit)")
                        .pantopusTextStyle(.small)
                        .foregroundStyle(Theme.Color.handyman)
                }
                HStack(spacing: Spacing.s1) {
                    Icon(.clock, size: 12, color: Theme.Color.handyman)
                    Text(bid.eta)
                        .pantopusTextStyle(.caption)
                        .foregroundStyle(Theme.Color.appText)
                }
            }
            Spacer(minLength: Spacing.s0)
            expiryPill
        }
        .accessibilityElement(children: .combine)
        .accessibilityLabel("Bid amount $\(bid.amount) \(bid.unit), \(bid.eta)")
    }

    @ViewBuilder private var expiryPill: some View {
        let text = isAccepted ? "Locked in" : bid.expires
        let tint = isAccepted ? Theme.Color.success : Theme.Color.handyman
        let bg = isAccepted ? Theme.Color.successBg : Theme.Color.appSurface
        if !text.isEmpty {
            Text(text)
                .font(.system(size: 10, weight: .heavy))
                .foregroundStyle(tint)
                .padding(.horizontal, Spacing.s2)
                .padding(.vertical, Spacing.s1)
                .background(bg)
                .overlay(
                    RoundedRectangle(cornerRadius: Radii.pill)
                        .stroke(tint.opacity(0.4), lineWidth: 1)
                )
                .clipShape(RoundedRectangle(cornerRadius: Radii.pill))
                .fixedSize()
        }
    }

    private var messageBlock: some View {
        VStack(alignment: .leading, spacing: Spacing.s1) {
            Text("THEIR MESSAGE")
                .pantopusTextStyle(.overline)
                .foregroundStyle(Theme.Color.handyman)
            VStack(alignment: .leading, spacing: Spacing.s1) {
                ForEach(Array(bid.message.enumerated()), id: \.offset) { _, paragraph in
                    Text("“\(paragraph)”")
                        .pantopusTextStyle(.small)
                        .foregroundStyle(Theme.Color.appText)
                        .fixedSize(horizontal: false, vertical: true)
                }
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/mailbox/item_detail/bodies/components/BidCard.kt
@file:Suppress("MagicNumber", "PackageNaming")

package app.pantopus.android.ui.screens.mailbox.item_detail.bodies.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.data.api.models.mailbox.v2.GigDetailDto
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon
import app.pantopus.android.ui.theme.PantopusIconImage
import app.pantopus.android.ui.theme.PantopusTextStyle
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * The bid headline card for the Gig mail body (A17.6) — large amount on the
 * left, unit + ETA, an expiry / "Locked in" pill on the right, and the
 * bidder's multi-line message below. Tinted with the gig accent (orange)
 * so it reads as the focal surface.
 */
@Composable
fun BidCard(
    bid: GigDetailDto.Bid,
    isAccepted: Boolean,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.xl))
                .background(PantopusColors.handyman.copy(alpha = 0.06f))
                .border(1.5.dp, PantopusColors.handyman.copy(alpha = 0.4f), RoundedCornerShape(Radii.xl))
                .testTag("gigBidCard"),
    ) {
        AmountRow(bid, isAccepted)
        Box(
            modifier =
                Modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(PantopusColors.handyman.copy(alpha = 0.3f)),
        )
        MessageBlock(bid)
    }
}

@Composable
private fun AmountRow(
    bid: GigDetailDto.Bid,
    isAccepted: Boolean,
) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(Spacing.s3),
        horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(Spacing.s1),
        ) {
            Text("BID AMOUNT", style = PantopusTextStyle.overline, color = PantopusColors.handyman)
            Row(verticalAlignment = Alignment.Bottom, horizontalArrangement = Arrangement.spacedBy(Spacing.s1)) {
                Text(
                    "$${bid.amount}",
                    fontSize = 34.sp,
                    fontWeight = FontWeight.Black,
                    color = PantopusColors.appTextStrong,
                )
                Text(
                    "· ${bid.unit}",
                    style = PantopusTextStyle.small,
                    color = PantopusColors.handyman,
                    modifier = Modifier.padding(bottom = Spacing.s1),
                )
            }
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(Spacing.s1),
            ) {
                PantopusIconImage(
                    icon = PantopusIcon.Clock,
                    contentDescription = null,
                    size = Radii.lg,
                    tint = PantopusColors.handyman,
                )
                Text(bid.eta, style = PantopusTextStyle.caption, color = PantopusColors.appText)
            }
        }
        ExpiryPill(bid = bid, isAccepted = isAccepted)
    }
}

@Composable
private fun ExpiryPill(
    bid: GigDetailDto.Bid,
    isAccepted: Boolean,
) {
    val text = if (isAccepted) "Locked in" else bid.expires
    if (text.isEmpty()) return
    val tint: Color = if (isAccepted) PantopusColors.success else PantopusColors.handyman
    val background: Color = if (isAccepted) PantopusColors.successBg else PantopusColors.appSurface
    Text(
        text = text,
        fontSize = 10.sp,
        fontWeight = FontWeight.Black,
        color = tint,
        modifier =
            Modifier
                .clip(RoundedCornerShape(Radii.pill))
                .background(background)
                .border(1.dp, tint.copy(alpha = 0.4f), RoundedCornerShape(Radii.pill))
                .padding(horizontal = Spacing.s2, vertical = Spacing.s1),
    )
}

@Composable
private fun MessageBlock(bid: GigDetailDto.Bid) {
    Column(
        modifier = Modifier.fillMaxWidth().padding(Spacing.s3),
        verticalArrangement = Arrangement.spacedBy(Spacing.s1),
    ) {
        Text("THEIR MESSAGE", style = PantopusTextStyle.overline, color = PantopusColors.handyman)
        bid.message.forEach { paragraph ->
            Text("“$paragraph”", style = PantopusTextStyle.small, color = PantopusColors.appText)
        }
    }
}
